### PR TITLE
[onert] Fix value error when training MaxPool2D

### PR DESCRIPTION
**Pull request by zetwhite**
This fixes value error when training a model with MaxPool2D.
It fixs a error in arguments passing order when configuring MaxPool layer.
This PR also adds related test cases to gurantee that this fix is correct.

ONE-DCO-1.0-Signed-off-by: seunghui youn <sseung.youn@samsung.com>
Co-authored-by: ragmani <ragmani0216@gmail.com>

### CODE REVIEW

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc [hunk 1]
```diff
@@ -0,0 +1,246 @@
...
+Epoch 2/5
+1/1 [==============================] - 0s 2ms/step - loss: 28.6837 - categorical_accuracy: 0.6667
+Epoch 3/5
+1/1 [==============================] - 0s 2ms/step - loss: 26.1765 - categorical_accuracy: 0.7778
+Epoch 4/5
+1/1 [==============================] - 0s 2ms/step - loss: 24.0089 - categorical_accuracy: 0.7778
+Epoch 5/5
+1/1 [==============================] - 0s 2ms/step - loss: 22.0964 - categorical_accuracy: 0.8889
+*/
+TEST_F(GenModelTrain, NonTrainableOps_Conv2D_MaxPool2D)
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:42]:
```bash
➜ ./nnfw_api_gtest --gtest_filter='GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D' 
Note: Google Test filter = GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D
[==========] Running 1 test from 1 test suite.
[----------] Global test environment set-up.
[----------] 1 test from GenModelTrain
[ RUN      ] GenModelTrain.NonTrainableOps_Conv2D_MaxPool2D
[1]    3801568 segmentation fault (core dumped)  ./nnfw_api_gtest 
```

File: runtime/onert/backend/train/ops/PoolLayer.cc [hunk 2]
```diff
@@ -45,7 +45,7 @@ class MaxPool2D final : public TrainingKernelRegistry
   std::unique_ptr<Tensor> _arg_max_index;
 
 public:
-  MaxPool2D(const uint32_t paddingLeft, const uint32_t, const uint32_t, const uint32_t paddingTop,
+  MaxPool2D(const uint32_t paddingLeft, const uint32_t, const uint32_t paddingTop, const uint32_t,
```
Comment (zetwhite) [inline runtime/onert/backend/train/ops/PoolLayer.cc:48]: @ragmani This line was a problem. :smiling_face_with_tear:

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc [hunk 3]
```diff
@@ -0,0 +1,172 @@
...
+
+auto getDefaultPool2DOptions()
+{
+  return std::tuple<circle::Padding, int, int, int, int, circle::ActivationFunctionType>(
+    circle::Padding_SAME, 1, 1, 1, 1, circle::ActivationFunctionType_NONE);
+}
+
+} // namespace
+
+TEST_F(GenModelTrain, NonTrainableOps_Conv2D_MaxPool2D)
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:30]:
```

/*
Epoch 1/5
1/1 [==============================] - 0s 152ms/step - loss: 31.6667 
Epoch 2/5
1/1 [==============================] - 0s 2ms/step - loss: 28.6837
Epoch 3/5
1/1 [==============================] - 0s 2ms/step - loss: 26.1765
Epoch 4/5
1/1 [==============================] - 0s 2ms/step - loss: 24.0089
Epoch 5/5
1/1 [==============================] - 0s 2ms/step - loss: 22.0964
*/
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+                        ));
+
+    _context->setBackends({"train"});
+    _context->setEpoch(4);
+
+    SUCCEED();
+  }
+}
+
+TEST_F(GenModelTrain, NonTrainableOps_Conv2D_MaxPool2D_Depth1_Filter2)
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:75]:
```
/*
Epoch 1/5
1/1 [==============================] - 0s 255us/step - loss: 31.6667
Epoch 2/5
1/1 [==============================] - 0s 120us/step - loss: 25.9453
Epoch 3/5
1/1 [==============================] - 0s 133us/step - loss: 15.4067
Epoch 4/5
1/1 [==============================] - 0s 131us/step - loss: 8.4666
Epoch 5/5
1/1 [==============================] - 0s 152us/step - loss: 6.4745
*/
```

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc [hunk 5]
```diff
@@ -0,0 +1,172 @@
...
+                        ));
+
+    _context->setBackends({"train"});
+    _context->setEpoch(4);
+
+    SUCCEED();
+  }
+}
+
+TEST_F(GenModelTrain, NonTrainableOps_Conv2D_MaxPool2D_Depth2_Filter2)
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:123]:
```
/*
Epoch 1/5
1/1 [==============================] - 0s 150ms/step - loss: 31.6667
Epoch 2/5
1/1 [==============================] - 0s 2ms/step - loss: 27.8823
Epoch 3/5
1/1 [==============================] - 0s 2ms/step - loss: 16.9743
Epoch 4/5
1/1 [==============================] - 0s 2ms/step - loss: 9.3556
Epoch 5/5
1/1 [==============================] - 0s 2ms/step - loss: 5.8163
*/
```

File: runtime/onert/backend/train/ops/PoolLayer.cc (hunk 2 above)
Comment (ragmani) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: Are you going to work on it with another PR? Or here?

Comment (ragmani) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: Are you going to work on it at another PR? Or here?

Comment (zetwhite) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: I'll make another PR to fix `stirde >1`. 
But I don't know a reason why `stride>1` gets a wrong value. So it might take some time.

Comment (ragmani) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: It's not urgent :)

**Review by ragmani — APPROVED** [review]
LGTM

File: runtime/onert/backend/train/ops/PoolLayer.cc (hunk 2 above)
Comment (zetwhite) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: I hope this PR is merged as it is, after getting reviewed.

Comment (ragmani) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: Ah... mnist model of hugging face may have max_pool with `stride>1`, but I'm not sure.

Comment (zetwhite) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: @ragmani, @jyoungyun  

`stride > 1` cases also work well! 

Yesterday, I didn't set the epoch properly, that's why the test failed. 
Now test with stride =2 also works well with properlly set epoch.

File: tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc [hunk 6]
```diff
@@ -0,0 +1,172 @@
...
+
+    _context->setBackends({"train"});
+    _context->setEpoch(4);
+
+    SUCCEED();
+  }
+}
+
+// TODO : Add test case to check stride supporting
+// TODO : Add more negative test cases
```
Comment (zetwhite) [inline tests/nnfw_api/src/GenModelTests/nontrainable_op_trains/MaxPool2D.test.cc:172]: I'll add this test cases in next PR.

File: runtime/onert/backend/train/ops/PoolLayer.cc (hunk 2 above)
Comment (ragmani) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: I checked kaggle model with this commit. The loss values ​​of `tensorflow` and `onert_train` were exactly the same. It proves not only this commit but also "stride > 1 cases also work well" is right.
- `tensorflow` result (mse, sgd, learning_rate=1.0f)
```
Epoch 1/5
10/10 [==============================] - 0s 6ms/step - loss: 0.0882 - categorical_accuracy: 0.9000
Epoch 2/5
10/10 [==============================] - 0s 5ms/step - loss: 0.0835 - categorical_accuracy: 1.0000
Epoch 3/5
10/10 [==============================] - 0s 5ms/step - loss: 0.0772 - categorical_accuracy: 1.0000
Epoch 4/5
10/10 [==============================] - 0s 5ms/step - loss: 0.0688 - categorical_accuracy: 1.0000
Epoch 5/5
10/10 [==============================] - 0s 5ms/step - loss: 0.0583 - categorical_accuracy: 1.0000
```

- `onert_train` result (mse, sgd, learning_rate=1.0f)
```
Epoch 1/5 - time: 19.255ms/step - loss: [0] 0.0882
Epoch 2/5 - time: 19.089ms/step - loss: [0] 0.0835
Epoch 3/5 - time: 18.987ms/step - loss: [0] 0.0772
Epoch 4/5 - time: 19.199ms/step - loss: [0] 0.0688
Epoch 5/5 - time: 19.255ms/step - loss: [0] 0.0583
```

Comment (zetwhite) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: If possible, it would be helpful to add a `mnist` model in CI to make sure that each PR is not broke the basic training feature.

Comment (ragmani) [reply runtime/onert/backend/train/ops/PoolLayer.cc:48]: I agree with @zetwhite

**Review by hseok-oh — APPROVED** [review]
